client/src/main.rs

```diff
@@ -563,6 +571,36 @@ fn add_cidr(interface: &InterfaceName, opts: AddCidrOpts) -> Result<(), Error> {
     Ok(())
 }
 
+fn delete_cidr(interface: &InterfaceName, opts: DeleteCidrOpts) -> Result<(), Error> {
+    let InterfaceConfig { server, .. } = InterfaceConfig::from_interface(interface)?;
+    println!("Fetching CIDRs");
+    let api = Api::new(&server);
+    let cidrs: Vec<Cidr> = api.http("GET", "/admin/cidrs")?;
+    let peers: Vec<Peer> = api.http("GET", "/admin/peers")?;
+
+    let cidr_id = prompts::delete_cidr(&cidrs, &opts)?;
+
+    if cidrs.iter().any(|cidr| {
+        if let Some(parent_id) = cidr.contents.parent {
+            cidr_id == parent_id
+        } else {
+            false
+        }
```

Comment:
seems like a good time for `matches!`
```suggestion
        matches!(cidr.contents.parent, Some(parent_id) if parent_id == cidr_id)
```

Reply:
Looks great, I wasn't aware of this macro!